components/suggestion/suggestion_box.jsx

```diff
@@ -188,6 +188,14 @@ export default class SuggestionBox extends React.Component {
     }
 
     componentDidUpdate(prevProps) {
+        const {value} = this.props;
+
+        // Post was just submitted, update pretext property.
+        if (value === '' && this.pretext !== value) {
```

Comment:
I'm thinking there may be a better way of letting the component know that it needs to reset itself, though this is the cleanest solution I found. There isn't much context to go off of after the post is submitted, other than incoming props.

Reply:
This is a scenario where I imagine storing the state in Redux and handling the transition across actions would give us the flexibility to "fix" the pretext. But that's a much bigger change.

I have no concerns with this, though we should keep an eye out for regressions.